flexget/plugins/input/anidb_feed.py

```diff
@@ -0,0 +1,479 @@
+from builtins import *  # noqa pylint: disable=unused-import, redefined-builtin
+
+import logging
+import re
+from xml.dom import minidom
+
+import feedparser
+from time import mktime
+from dateutil.parser import parse as dateutil_parse
+from dateutil.tz import tz
+from datetime import datetime
+
+from flexget.config_schema import one_or_more
+
+from flexget.utils.qualities import Quality
+from past.builtins import basestring
+from flexget.utils.tools import str_to_boolean
+
+from flexget import plugin
+from flexget.event import event
+from flexget.utils.cached_input import cached
+from flexget.entry import Entry
+from flexget.utils.requests import TimedLimiter
+
+
+log = logging.getLogger('anidb_feed')
+
+HEADER = {'User-agent': 'Mozilla/5.0'}  # anidb needs valid header or we get '403 Client Error'
+LIMITER = TimedLimiter('anidb.net', '2 seconds')  # default anidb api limit
+
+
+NAMESPACE_NAME = 'xhtml'
+NAMESPACE_URL = 'http://www.w3.org/1999/xhtml'
+NAMESPACE_TAGNAME = 'dl'
+NAMESPACE_PREFIX_MAIN = 'anidb_'
+NAMESPACE_PREFIX = 'anidb_feed_'
+
+# NOTE: all lowercase only
+NAMESPACE_ATTRIBUTE_MAP = {
+    'added': datetime,
+    'source': basestring,
+    'resolution': basestring,
+    'crc_status': basestring,
+    'language': basestring,
+    'group': basestring,
+    'subtitle_language': basestring,
+    'priority': basestring,
+    'quality': basestring,
+    'size': basestring
+}
+
+VALIDATION_LIST = [
+    'source',
+    'resolution',
+    'group',
+    'size'
+]
+
+# FIXME: not 100% if this is correct or move directly to qualities.py?
+ANIDB_SOURCES_MAP = {
+    'camcorder': 'cam',
+    'TV': 'tvrip',
+    'DTV': 'sdtv',
+    'HDTV': 'hdtv',
+    'VHS': 'tvrip',
+    'VCD': 'tvrip',
+    'SVCD': 'sdtv',
+    'LD': 'sdtv',
+    'DVD': 'dvdrip',
+    'HKDVD': 'dvdrip',
+    'HD-DVD': 'dvdrip',
+    'Blu-ray': 'bluray',
+    'www': 'webdl'
+}
+
+ANIDB_EXTRA_TYPES = [
+    'special',
+    'op-ending',
+    'trailer-promo'
+]
+
+
+def convert_to_naive_utc(valuestring):
+    parsed_date = None
+    try:
+        parsed_date = dateutil_parse(valuestring, fuzzy=True)
+        try:
+            parsed_date = parsed_date.astimezone(tz.tzutc()).replace(tzinfo=None)
+        except ValueError:
+            parsed_date = parsed_date.replace(tzinfo=None)
+    except ValueError as ex:
+        log.warning('Invalid datetime field format: %s error: %s' % (valuestring, ex))
+    except Exception as ex:
+        log.trace('Unexpected datetime field: %s error: %s' % (valuestring, ex))
+    return parsed_date
+
+
+def convert_to_number(valuestring):
+    number = None
+    try:
+        number = int(valuestring)
+    except ValueError:
+        try:
+            number = float(valuestring)
+        except ValueError:
+            log.trace('Invalid number field: %s' % valuestring)
+    except Exception as ex:
+        log.trace('Invalid number field: %s error: %s' % (valuestring, ex))
+    return number
+
+
+def _debug_dump_entry(entry):
```

Comment:
Not really, `dump()` uses too short left anchor, which results in hard readable line entries and has no ability to output the actual field type, which i was interested in this case.